### DISCUSSION TITLE
Copilot Labs Issue: Brushes

### DISCUSSION THREAD

**Discussion by arudaev** [community/community#49683] [Copilot Conversations]
### Select Topic Area

Bug

## Description
Recently (within the last week), Copilot Labs has been experiencing an issue with its Brushes feature. When I attempt to use any of the Brushes functions (e.g. "Make more readable," "Clean," or "Debug"), sections of the code are deleted without any effect. Initially, I thought the problem was due to documentation/comments within functions, but this was not the case. Additionally, when attempting to use **Brushes, either the last and half of the before the last line are deleted**, or **everything except the first line and half of the second line.** This issue occurs when working with Python, C++, and C#, don't know about JavaScript nor TypeScript.

## Steps to Reproduce

---

**Before using brushes (with comments):**
![image](https://user-images.githubusercontent.com/66517969/224265950-e4aebafc-b0a6-49f0-95b7-ae580755aeee.png)

**After using the brushes (with comments):**
![image](https://user-images.githubusercontent.com/66517969/224266302-caaa0f13-5ad5-4855-9556-4927e5367294.png)

---

**Before using brushes (no comments):**
![image](https://user-images.githubusercontent.com/66517969/224266429-ed1dbea9-5870-47cd-a697-d7d7f1156889.png)

**After using brushes (no comments):**
![image](https://user-images.githubusercontent.com/66517969/224266562-a4ff5253-afbf-493f-9956-e443350ce8a5.png)

## Additional information
OS: Win11
IDE/TextEditor: VS Code
Extension: GitHub Copilot Labs v0.12.791

**Comment by l-t-m-f** [comment]
The same problem is happening to me. When I ask it to List steps, it deletes my code LOL.

**Comment by stckcrsh** [comment]
I am getting the same thing with `Readable`, `Add Types`, `Fix Bug`, `Debug`, `List Steps`, `Make Robust`, `Chunk`

`Document` kinda works and creates a good comments but still cuts out most of the function.

`Clean` has worked every time for me

**Comment by alex-reilly-pronto** [comment]
I'm seeing the same thing with the custom brush

**Comment by talkingnews** [comment]
Same here - For example, if I use the "readable" brush, it cuts my code off after` fs.mkdirSync(` and if I use the "Fix bug" brush, it cuts my code off further down after `fs.writeFileSync(`  



It is definitely "seeing" the whole function, because it shows the whole function in the test window and will generate a test.

I thought it might be eslint or AWS CodeWhisperer or VsCode JS formatting in a race condition with CoPilot's result, but I disabled of them, and in the Copilot Labs window, it has `Invalid streamingData,joinedText:[` (then the code, cut short) and then `] not a substring of joinedTokens:[]` 

```
GitHub Copilot Labs v0.14.884
GitHub Copilot Nightly v1.88.150
VSCode Version: 1.80.0-insider (user setup)
Commit: 6cd5ac180a6362b45b415a624f62173caf7428a1
Date: 2023-06-09T18:28:20.007Z
Electron: 22.5.5
Chromium: 108.0.5359.215
Node.js: 16.17.1
V8: 10.8.168.25-electron.0
OS: Windows_NT x64 10.0.19045
```
